handler/src/main/java/io/netty/handler/ssl/SslContextBuilder.java

```diff
@@ -423,6 +447,24 @@ public SslContextBuilder keyManager(KeyManagerFactory keyManagerFactory) {
         return this;
     }
 
+    /**
+     * A single key manager managing the identity information of this host.
+     * This is helpful when custom implementation of {@link KeyManager} is needed.
+     * Internally, a wrapper of {@link KeyManagerFactory} that only produces this specified
+     * {@link KeyManager} will be created, thus all the requirements specified in
+     * {@link #keyManager(KeyManagerFactory keyManagerFactory)} also apply here.
+     */
+    public SslContextBuilder keyManager(KeyManager keyManager) {
+        if (forServer) {
```

Comment:
The `forServer == false` flow looks wrong. If `null` is passed, then `KeyManagerFactoryWrapper` will throw (and we probably want the `keyManagerFactory` to be `null` in that case).

Reply:
Thanks for pointing that out. Updated accordingly. Could you please take a second look?